src/security/PeerConnector.cc

```diff
@@ -394,6 +395,14 @@ Security::PeerConnector::handleNegotiateError(const int ret)
         return;
 
     case SSL_ERROR_WANT_WRITE:
+        if (needsValidationCallouts()) {
+            typedef NullaryMemFunT<Security::PeerConnector> cbDialer;
+            AsyncCall::Pointer resumeCall = JobCallback(83, 5,
+                                            cbDialer, this,
+                                            Security::PeerConnector::noteWantWrite);
+            suspendNegotiation(resumeCall);
+        }
+
         noteWantWrite();
```

Comment:
AFAICT, noteWantWrite() calls functions like checkForPeekAndSplice() that feel like making progress rather than just remembering the need to write later. Thus, the combination of "suspend" and "noteWantWrite" looks self-contradictory to me. @chtsanti, please double check that we are not missing a `return` after suspendNegotiation().

Reply:
Fixed in commit a30a6415ad8